tests/validation/tests/v3_api/test_group_grb.py

```diff
@@ -1,110 +1,449 @@
+        validate_permission_list_cluster(user, target_num)
+    # check that user nestedtestuser1 can not list clusters
+    validate_permission_list_cluster("nestedtestuser1", 0)
+    # delete the ggrb
+    admin_c.delete(ggrb)
+    # check that users in the group can not list clusters
+    for user in users:
+        validate_permission_list_cluster(user, 0)
+
+
+def test_ggrb_3(remove_resource):
+    """ test that after editing the global role, users'
+    permissions reflect the changes
+
+    Steps:
+    - users in nestgroup1 cannot list clusters or create catalogs
+    - create a custom global role gr1 that permits creating catalogs
+    - create the global group role binding ggrb1 to bind gr1 to nestgroup1
+    - users in nestgroup1 can create catalogs now
+    - edit gr1 to remove the permission of creating catalogs, add the
+    permission of listing clusters
+    - users in nestgroup1 cannot create catalogs, but can list clusters
+    """
+
+    target_group_name = "nestgroup1"
+    users = get_direct_user_by_group(target_group_name)
+    if NESTED_GROUP_ENABLED:
+        users = get_user_by_group(target_group_name)
+    # check that users can not create catalogs or list clusters
+    for user in users:
+        validate_permission_create_catalog(user, False)
+        validate_permission_list_cluster(user, 0)
+    # create a custom global role that permits create catalogs
+    admin_c = get_admin_client()
+    template = generate_template_global_role(name=random_name(),
+                                             template=TEMPLATE_MANAGE_CATALOG)
+    gr = admin_c.create_global_role(template)
+    remove_resource(gr)
+    groups = search_ad_groups(target_group_name, ADMIN_TOKEN)
+    assert len(groups) == 1
+    ggrb = get_admin_client().create_global_role_binding(
+        globalRoleId=gr["id"], groupPrincipalId=groups[0]["id"])
+    # check that users can create catalogs now, but not list clusters
+    for user in users:
+        validate_permission_list_cluster(user, 0)
+        catalog = validate_permission_create_catalog(user, True)
+        remove_resource(catalog)
+    # edit the global role
+    rules = [
+        {
+            "type": "/v3/schemas/policyRule",
+            "apiGroups": [
+                "management.cattle.io"
+            ],
+            "verbs": [
+                "get",
+                "list",
+                "watch"
+            ],
+            "resources": [
+                "clusters"
+            ]
+        }
+    ]
+    admin_c.update(gr, rules=rules)
+    target_num = len(admin_c.list_cluster().data)
+    # check that users can list clusters, but not create catalogs
+    for user in users:
+        validate_permission_create_catalog(user, False)
+        validate_permission_list_cluster(user, target_num)
+    # delete the ggrb
+    admin_c.delete(ggrb)
+    for user in users:
+        validate_permission_list_cluster(user, 0)
+
+
+@if_test_rbac
+def test_rbac_ggrb_list_cluster_owner():
```

Comment:
Can you parameterize these tests? Instead of a completely different test for each role: have one test per operation (create, delete, list), and pass the role & expected output for scenarios as parameters (See max's new Istio tests)

Reply:
done